featurex/diagnostics/__init__.py

```diff
@@ -0,0 +1,123 @@
+        self.results['Variances'] = variances(self.data[cols])
+
+
+    def show(self, stdout=True, plot=False):
+        """
+        Displays diagnostics to the user
+
+        Args:
+            stdout (bool): print results to the console
+            plot (bool): use Seaborn to plot results
+        """
+        if stdout:
+            print('Collinearity summary:')
+            print(pd.concat([self.results['Eigenvalues'],
+                            self.results['ConditionIndices'], 
+                            self.results['VIFs'],
+                            self.results['CorrelationMatrix']],
+                            axis=1))
+
+            print('Outlier summary:')
+            print(self.results['RowMahalanobisDistances'])
+            print(self.results['ColumnMahalanobisDistances'])
+
+            print('Validity summary:')
+            print(self.results['Variances'])
+        
+        if plot:
+            for key, result in self.results.items():
+                if key == 'CorrelationMatrix':
+                    ax = plt.axes()
+                    sns.heatmap(result, cmap='Blues', ax=ax)
+                    ax.set_title(key)
+                    sns.plt.show()
+                else:
+                    result.plot(kind='bar', title=key)
+                    plt.show()
+
+
+    def flag(self, diagnostic, thresh=None):
+        """
+        Returns indices of diagnostic that satisfy (return True from) the 
+        threshold predicate. Will use class-level default threshold if 
+        None provided.
+
+        Args:
+            diagnostic (str): name of the diagnostic
+            thresh (func): threshold function (boolean predicate) to apply to 
+            each element
+        """
+        if thresh is None:
+            thresh = self.defaults[diagnostic]
+
+        result = self.results[diagnostic]
+        if isinstance(result, pd.DataFrame):
+            if diagnostic == 'CorrelationMatrix':
+                result = result.copy()
+                np.fill_diagonal(result.values, 0)
+            return result.applymap(thresh).sum().nonzero()[0]
+        else:
+            return result.apply(thresh).nonzero()[0]
+
+
+    def flag_all(self, thresh_dict={}):
+        """
+        Returns indices of (rows, columns) that satisfy flag() on any diagnostic
+        Uses user-provided thresholds in thresh_dict
+
+        Args:
+            thresh_dict (dict): dictionary of diagnostic->threshold functions
+        """
+        row_idx = set()
+        col_idx = set()
+        for diagnostic in self.results:
+            if diagnostic in thresh_dict:
+                flagged = self.flag(diagnostic, thresh_dict[diagnostic])
+            else:
+                flagged = self.flag(diagnostic)
+
+            if diagnostic == 'RowMahalanobisDistances':
+                row_idx = row_idx.union(flagged)
+            else:
+                col_idx = col_idx.union(flagged)
+
+        return row_idx, col_idx
```

Comment:
We probably don't want to return sets, because I don't think they can be used in indexing arrays. Maybe convert to (sorted) lists on the return.